code/game/objects/items.dm

```diff
@@ -187,6 +187,8 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
 
 	var/canMouseDown = FALSE
 
+	/// Item state of tank holder when inside tank holder. null makes it unable to enter a tank holder.
+	var/tank_holder_icon_state = null
```

Comment:
there has to be a better way other than having this on literally every item in the game lol

Reply:
Not really sure what the best course of action is here.

Should I make a component? Is it really better than making a proc, let's say?

Reply:
If nothing else could this just be undefined rather than null on every object? And have it be defined only for objects that can be in a tank holder?

Reply:
I've added components already

Reply:
Please take a look to see if that is ok @ShizCalev